pkg/kinflate/configmapandsecret/configmap_secret_test.go

```diff
@@ -252,84 +179,36 @@ func TestConstructConfigMap(t *testing.T) {
 	}
 }
 
-func TestConstructTLSSecret(t *testing.T) {
-	type testCase struct {
-		description string
-		input       manifest.TLSSecret
-		expected    *corev1.Secret
-	}
-
-	testCases := []testCase{
-		{
-			description: "construct secret from tls",
-			input: manifest.TLSSecret{
-				Name:     "tlsSecret",
-				CertFile: "../examples/simple/instances/exampleinstance/secret/tls.cert",
-				KeyFile:  "../examples/simple/instances/exampleinstance/secret/tls.key",
-			},
-			expected: makeTLSSecret("tlsSecret"),
+func TestConstructSecret(t *testing.T) {
+	secret := manifest.SecretGenerator{
+		Name: "secret",
+		Commands: map[string]string{
+			"DB_USERNAME": "printf admin",
+			"DB_PASSWORD": "printf somepw",
 		},
+		Type: "Opaque",
 	}
-
-	for _, tc := range testCases {
-		cm, err := makeTlsSecret(tc.input)
-		if err != nil {
-			t.Fatalf("unepxected error: %v", err)
-		}
-		if !reflect.DeepEqual(*cm, *tc.expected) {
-			t.Fatalf("in testcase: %q updated:\n%#v\ndoesn't match expected:\n%#v\n", tc.description, *cm, tc.expected)
-		}
+	cm, err := makeSecret(secret, ".")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
 	}
-}
-
-func TestConstructGenericSecret(t *testing.T) {
-	type testCase struct {
-		description string
-		input       manifest.GenericSecret
-		expected    *corev1.Secret
+	expected := makeTestSecret("secret")
+	if !reflect.DeepEqual(*cm, *expected) {
+		t.Fatalf("%#v\ndoesn't match expected:\n%#v", *cm, *expected)
 	}
+}
 
-	testCases := []testCase{
-		{
-			description: "construct secret from env",
-			input: manifest.GenericSecret{
-				Name: "envSecret",
-				DataSources: manifest.DataSources{
-					EnvSource: "../examples/simple/instances/exampleinstance/configmap/app.env",
-				},
-			},
-			expected: makeEnvSecret("envSecret"),
-		},
-		{
-			description: "construct secret from file",
-			input: manifest.GenericSecret{
-				Name: "fileSecret",
-				DataSources: manifest.DataSources{
-					FileSources: []string{"../examples/simple/instances/exampleinstance/configmap/app-init.ini"},
-				},
-			},
-			expected: makeFileSecret("fileSecret"),
-		},
-		{
-			description: "construct secret from literal",
-			input: manifest.GenericSecret{
-				Name: "literalSecret",
-				DataSources: manifest.DataSources{
-					LiteralSources: []string{"a=x", "b=y"},
-				},
-			},
-			expected: makeLiteralSecret("literalSecret"),
+func TestFailConstructSecret(t *testing.T) {
+	secret := manifest.SecretGenerator{
+		Name: "secret",
+		Commands: map[string]string{
+			"FAILURE": "false", // This will fail.
```

Comment:
Nice comment.  It's not obvious that there's a command called false that always fails :)

Does the coreutils on a Mac have this?  I don't care.

Reply:
Yeah, that's kind of mandatory, otherwise you can't do `if false; then ...` which is obviously super useful =)
FYI, there is also `true` that always succeeds :-)